### ISSUE TITLE
[usdExport] "exportDisplayColor" results in black for materials with textures

### ISSUE THREAD

**Issue by nxkb** [Autodesk/maya-usd#1705]
**Describe the bug**
When exporting display colors for a shape with an assigned material that is using a texture (file node), the result is black rather than the default color of the file node.

Basically the export code in `/lib/mayaUsd/utils/util.cpp::_getMayaShadersColor()` will ask the color attribute for its color, but if it's connected it will just return black. I propose that it would be more reasonable to lookup the default color of that file node so that artists would have some way to set a color (say "light brown" for a "sand" texture). 

**Steps to reproduce**
1. Create a cube and assign a new lambert material
2. Connect a file node to the 'color' attribute and set file node default color to (1.0, 0.0, 0.0)
3. select your cube and Export Selected to `test.usd`
4. Witness `/root/pCube1.displayColor == (0.0, 0.0, 0.0)` in usdview.

**Expected behavior**
`pCube1.displayColor == (1.0, 0.0, 0.0)`

**Impact**
Since this is case, artists have no way to set the displayColor AND a preview shader texture (or just keep the maya shader they need for other workflows). They have to disconnect the texture, set the color, export, and then reconnect their texture to get the right behavior. We currently have a warning in place to notify artists, but it would be nice to be able to grab the more appropriate color and leave everything connected.

**Workaround**
Assign a new (lambert) material with the default color in "color" OR disconnect the texture, set the color, export, and then reconnect the texture. 

**Specs:**
 - OS & version [CentOS]
 - Compiler & version [gcc 6.3.1]
 - Maya version [Maya 2020]
 - Maya USD commit SHA [dev at 0c906e8fd62fa83b57bad82638beabb855c7e7a5]

**Additional context**
Basically add the bold to the lookup for displayColor.
shape --> material --> surfaceShader.color **--> file1.defaultColor**

I'd like to at least start the conversation to see if this approach is agreeable to others. We don't quite rely on displayColor enough yet to jump on solving this, but I'd be happy to circle back when I have time and try to solve it if no one's done it yet.

**Comment by BigRoy** [comment]
For anyone looking into implementing this. The export display color flags ends up using this [`UsdMayaUtil::GetLinearShaderColor` logic](https://github.com/Autodesk/maya-usd/blob/b8d3fbc0b4045ecb464d5d0736bdba9edf0acc45/lib/mayaUsd/utils/util.cpp#L996-L1019) thus it seems mostly related to re-implementing this logic that actually retrieves the [Shader's color](https://github.com/Autodesk/maya-usd/blob/b8d3fbc0b4045ecb464d5d0736bdba9edf0acc45/lib/mayaUsd/utils/util.cpp#L973-L988) if a file node is connected to the input.

As such you'll likely need to only implement changes in the [`util.cpp` file](https://github.com/Autodesk/maya-usd/blob/dev/lib/mayaUsd/utils/util.cpp) to make this work.

**Comment by Ziethriel** [comment]
Not sure if this is related, but explicitly naming the color set "displayColor" instead of the default "colorSet1" has increased my compatibility with other applications (Substance Painter & Omniverse)

**Comment by ultra-sonic** [comment]
Hello Autodesk,
we just ran into the same issue and it would be great if the displayColor would be set to the default color that was set in the file node. ofc it would be even better if there was a flag to lookup the texture per vertex and store this - like a texture transfer
